homeassistant/helpers/typing.py

```diff
@@ -1,12 +1,39 @@
 """Typing Helpers for Home-Assistant."""
 from typing import Dict, Any
 
-import homeassistant.core
+# NOTE: NewType added to typing in 3.5.2 in June, 2016; Since 3.5.2 includes
+#       security fixes everyone on 3.5 should upgrade "soon"
+try:
+    from typing import NewType
+except ImportError:
+    NewType = None
+
+# HACK: mypy/pytype will import, other interpreters will not; this is to avoid
```

Comment:
There were at least two instances of a circular dependency. In the documentation there is a note on the future of this hack

> It is possible that in the future, mypy will change its dead code analysis and this trick will stop working. We will then offer an alternative, e.g. a constant defined by the typing module that is False at runtime but True while type checking.

python/typing, python/mypy and google/pytype will introduce a variable to indicate if it is being typechecked or not so instead of `if False:` it would be changed to something like `if __typing__:` or `if TYPE_CHECK:`. When that happens I we should be able to change it to that variable and remove the HACK documentation.